src/components/name-tag.js

```diff
@@ -0,0 +1,377 @@
+            typingAnimTime -= TYPING_ANIM_SPEED;
+          }
+        });
+      }
+      if (DEBUG) {
+        this.updateAvatarModelBBAA();
+        this.avatarBBAAHelper.matrixNeedsUpdate = true;
+        this.avatarBBAAHelper.updateMatrixWorld(true);
+      }
+    };
+  })(),
+
+  play() {
+    this.el.addEventListener("model-loading", this.onModelLoading);
+    this.el.addEventListener("model-loaded", this.onModelLoaded);
+    this.el.addEventListener("analyser-volume-updated", this.onAnalyserVolumeUpdated);
+    this.el.sceneEl.addEventListener("presence_updated", this.onPresenceUpdated);
+    window.APP.store.addEventListener("statechanged", this.onStateChanged);
+  },
+
+  pause() {
+    this.el.removeEventListener("model-loading", this.onModelLoading);
+    this.el.removeEventListener("model-loaded", this.onModelLoaded);
+    this.el.removeEventListener("analyser-volume-updated", this.onAnalyserVolumeUpdated);
+    this.el.sceneEl.removeEventListener("presence_updated", this.onPresenceUpdated);
+    window.APP.store.removeEventListener("statechanged", this.onStateChanged);
+  },
+
+  onPresenceUpdated({ detail: presenceMeta }) {
+    if (presenceMeta.sessionId === this.playerSessionId) {
+      this.updateFromPresenceMeta(presenceMeta);
+    }
+  },
+
+  updateFromPresenceMeta(presenceMeta) {
+    this.displayName = presenceMeta.profile.displayName;
+    this.identityName = presenceMeta.profile.identityName;
+    this.isRecording = !!(presenceMeta.streaming || presenceMeta.recording);
+    this.isOwner = !!(presenceMeta.roles && presenceMeta.roles.owner);
+    this.isTyping = !!presenceMeta.typing;
+    this.isHandRaised = !!presenceMeta.hand_raised;
+    this.updateNameTag();
+  },
+
+  updateDisplayName() {
+    if (this.displayName) {
+      this.nametagTextEl.addEventListener(
+        "text-updated",
+        () => {
+          this.size = this.nametagTextEl.components["text"].getSize();
+          this.size.x = Math.max(this.size.x, NAMETAG_MIN_WIDTH);
+          this.updateNameTag();
+        },
+        { once: true }
+      );
+      if (this.displayName.length > DISPLAY_NAME_LENGTH) {
+        this.displayName = this.displayName.slice(0, DISPLAY_NAME_LENGTH).concat("...");
+      }
+      this.nametagTextEl.setAttribute("text", {
+        value: this.displayName
+      });
+    }
+
+    if (this.identityName) {
+      if (this.identityName.length > DISPLAY_NAME_LENGTH) {
+        this.identityName = this.identityName.slice(0, DISPLAY_NAME_LENGTH).concat("...");
+      }
+      this.identityNameEl.setAttribute("text", { value: this.identityName });
+      this.identityNameEl.object3D.visible = this.el.sceneEl.is("frozen");
+    }
+  },
+
+  onModelLoading() {
+    this.tmpNametagVisible = false;
+    this.updateNameTagVisibility();
+    this.model = null;
+  },
+
+  onModelLoaded({ detail: { model } }) {
+    clearInterval(this.firstIkStepHandler);
+    const checkIkStep = () => {
```

Comment:
Hmm, this feels pretty messy. Do we actually care to wait for ik-controller's `hadFirstTick`? If we do it probably makes sense in this case for `ik-controller` to fire an event. 

Like the other comments in this PR, it call a function directly on nametag instead of firing an event, but in this case it feels odd to me for `ik-controller` to know about nametag. (But does not for example feel odd for the chat box to be involved in notifying about typing status)

Reply:
We need to wait for the first tick otherwise the BBAA size is not correct. Makes sense to use an event, updated.